src/pty.rs

```diff
@@ -0,0 +1,128 @@
+//! Create master and slave virtual pseudo-terminals (PTTYs)
+
+use std::ffi::CStr;
+use std::os::unix::io::RawFd;
+
+use libc;
+
+use {Error, fcntl, Result};
+
+/// Grant access to a slave pseudoterminal (see
+/// [grantpt(3)](http://man7.org/linux/man-pages/man3/grantpt.3.html))
+///
+/// `grantpt()` changezs the mode and owner of the slave pseudoterminal device corresponding to the
+/// master pseudoterminal referred to by `fd`. This is a necessary step towards opening the slave.
+#[inline]
+pub fn grantpt(fd: RawFd) -> Result<()> {
```

Comment:
Actually, `From` implies `Into`, but not the other way around.  If this newtype implements `Into<RawFd>` , that does not automatically create an implementation for `From<Newtype>` on `RawFd`.  Do you think that typing `.into()` is too much work to justify the free type checking on the pty functions?  If the newtype implements `Drop`, that could save some typing too, because you wouldn't have to automatically close the ptys.